mail_debrand/models/mail_thread.py

```diff
@@ -10,5 +10,15 @@ class MailThread(models.AbstractModel):
 
     def _replace_local_links(self, html, base_url=None):
         html = super()._replace_local_links(html, base_url=base_url)
-        html_debranded = self.env["mail.template"]._debrand_body(html)
+        lang = False
+        if {'default_template_id', 'default_model', 'default_res_id'} \
+                <= self.env.context.keys():
+            template = self.env["mail.template"].browse(
+                self.env.context['default_template_id'])
+            template_lang = template.lang or template._context.get('lang')
+            if template_lang:
+                lang = template._render_template(template_lang,
```

Comment:
> From [#736 (comment)](https://github.com/OCA/social/pull/736#discussion_r733688343):
> 
> > 
> 
> In order to debrand correctly the mail, you must translate the terms `using` and `Powered by` with the language set in the mail.template record **if exists, otherwise the language in context**.
> 
> See also https://github.com/odoo/odoo/blob/2be47637dd5cc48ca561ff85cf2d359dde0829ab/addons/mail/models/mail_template.py#L443

(from https://github.com/OCA/social/pull/736#discussion_r785845002)

I see, but I think the value in the context is `en_US` or `it_IT` or something similar, so looks to me like there is no need to call `_render_template` for it, that's why I asked:
> why is this rendering the language coming from the context?

and I don't understand how https://github.com/OCA/social/pull/736#discussion_r785845002 can answer that question, could you please rephrase?

Reply:
Got it.

Now it values `lang` variable directly with the value in the context if `template.lang` is `False`.